modules/apps/document-library/document-library-test/src/testIntegration/java/com/liferay/document/library/service/test/DLFileEntryLocalServiceTest.java

```diff
@@ -1253,6 +1258,54 @@ public void testUpdateDisplayDateExpirationDateReviewDate()
 		Assert.assertEquals(reviewDate, dlFileEntry.getReviewDate());
 	}
 
+	@Test
+	public void testUpdateFileEntryDynamicQueryTransactionInterval()
+		throws Exception {
+
+		for (int i = 0; i < 20; i++) {
+			_dlFileEntryLocalService.addFileEntry(
+				null, TestPropsValues.getUserId(), _group.getGroupId(),
+				_group.getGroupId(), DLFolderConstants.DEFAULT_PARENT_FOLDER_ID,
+				StringUtil.randomString(), ContentTypes.TEXT_PLAIN,
+				StringUtil.randomString(), StringUtil.randomString(),
+				StringPool.BLANK, StringPool.BLANK, -1, new HashMap<>(), null,
+				new ByteArrayInputStream(new byte[0]), 0, null, null, null,
+				ServiceContextTestUtil.getServiceContext(
+					_group.getGroupId(), TestPropsValues.getUserId()));
+		}
+
+		AtomicInteger atomicInteger = new AtomicInteger(0);
+
+		try {
+			_dlFileEntryLocalService.forEachFileEntry(
+				TestPropsValues.getCompanyId(),
+				dlFileEntry -> {
+					if (atomicInteger.incrementAndGet() == 20) {
+						throw new TestException();
+					}
+
+					dlFileEntry.setDescription("Updated");
+				},
+				0L, new String[] {ContentTypes.TEXT_PLAIN});
+		}
+		catch (RuntimeException runtimeException) {
+			if (!(runtimeException.getCause() instanceof TestException)) {
+				throw runtimeException;
+			}
```

Comment:
Why not catch the `TestException` directly? Also, the pattern to verify exceptions are thrown is:
```
try {
    _doSomethinThatShouldThrow();

    Assert.fail(...);
}
catch (...) {
}
```
Look for usages of `Assert.fail()` in the portal.

Reply:
As mentioned above, I did not catch TestException because it is wrapped inside a SystemException.

Regarding the Assert.fail I have changed it, thanks for the advice!